backends/alpm/pk-alpm-config.c

```diff
@@ -1014,7 +1014,10 @@ pk_alpm_config_configure_alpm (PkBackend *backend, PkAlpmConfig *config, GError
 	config->noupgrades = NULL;
 
 	/* alpm takes ownership */
-	alpm_option_set_noupgrades (handle, config->hookdirs);
+	alpm_list_t *j;
+	for(j = config->hookdirs; j; j = j->next) {
```

Comment:
can you put the alpm_list_t in the loop please, and squash all these down to a single commit pls

Reply:
I can squash no problem but I don't know how to put alpm_list_t inside the loop, it's the variable used to iterate, don't I need it declared beforehand?

Reply:
`for(alpm_list_t *j = config->hookdirs; `

Reply:
done